contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,94 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   // --------------------------------------------------------------------------
+   struct blacklist_manager {
+      blacklist_manager(std::vector<name>& blacklist) : _blacklist(blacklist) {}
+
+      void add(const std::vector<name>& patterns) const {
+         std::unordered_set<name, _hash> already;
+
+         for (auto n : _blacklist)
+            already.insert(n);
+
+         for (auto n : patterns) {
+            if (!already.contains(n))
+               _blacklist.push_back(n);
+         }
+      }
+
+      void remove(const std::vector<name>& patterns) const {
+         std::unordered_set<name, _hash> already;
```

Comment:
I wonder how much code is generated for the `unordered_set`. Might be worth using a sorted vector to reduce the amount of generated code.

Reply:
I don't mind doing this, but it is not obvious to me that it is better, with having to sort the vector and then doing a binary search into it.

Reply:
It is more about saving code bloat than performance. Not sure if code size is something to be concerned about, but the `eosio.system` contract is one of the bigger contracts.

Reply:
Since @arhag mentioned that there will likely be a very small number of names in the blacklist, maybe I should just get rid of the `blacklist_manager` and just search in the existing list for each name we want to add. It is quadratic, but if we have even 20 names it will be fast enough.

Reply:
removed the `blacklist_manager`: https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/f31f3cf29ced5bf3bed259701480aec39fe176d1